spec/lib/good_job/systemd_service_spec.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+# Creates a socket at a temp path that you can read from during a test.
+class TestSocket
+  attr_accessor :path
+
+  def initialize
+    @socket = Socket.new(:UNIX, :DGRAM, 0)
+    @path = nil
+    Dir::Tmpname.create("test_socket") do |filepath|
+      @path = filepath
+      socket_info = Addrinfo.unix(@path)
+      @socket.bind(socket_info)
+    end
+  end
+
+  def close
+    @socket.close
+    File.unlink(@path) if @path
+  end
+
+  def read(maxlen = 16)
+    @socket.recvfrom(maxlen)[0]
+  end
+end
+
+RSpec.describe GoodJob::SystemdService, skip_if_java: true do
```

Comment:
JRuby seems to have trouble binding to the socket (this only matters for tests, not the actual implementation, I think).

I noticed Puma also skips tests on JRuby, presumably for this reason. Not sure if switching to `UNIXSocket` would help, but I don’t really have more time to play around with this at the moment.

Reply:
This is good 👍🏻